argo-cd-apps/base/all-clusters/infra-deployments/tracing-workload-otel-collector/tracing-workload-tracing.yaml

```diff
@@ -10,28 +10,31 @@ spec:
         generators:
           - clusters:
               values:
-                sourceRoot: configs/enable-tekton-tracing
+                sourceRoot: components/tracing/otel-collector
                 environment: staging
                 clusterDir: ""
           - list:
               elements: []
   template:
     metadata:
-      name: enable-tekton-tracing-{{nameNormalized}}
+      name: tracing-workload-tracing-{{nameNormalized}}
     spec:
       project: default
       source:
-        path: configs/enable-tekton-tracing
+        path: '{{values.sourceRoot}}/{{values.environment}}/{{values.clusterDir}}'
         repoURL: https://github.com/redhat-appstudio/infra-deployments.git
```

Comment:
In order to support the deployment of the development environment, you should set the values of `repoUrl` and `targetRevision` using a kustomize component, simliar to argo-cd-apps/k-components/inject-infra-deployments-repo-details.

The values in the new kustomize component you are going to create should be modified by the `preview.sh` script (https://github.com/redhat-appstudio/infra-deployments/blob/main/hack/preview.sh#L125)

It's needed since when deploying a development env (locally or in CI) a fork and a branch other then main are used.

Reply:
I just walked through the code, it should be applied by the existing kustomize components.....

Reply:
And also verfied the behavior in my local dev environment.

Reply:
here https://github.com/redhat-appstudio/infra-deployments/pull/3917/files#diff-fc45d4db7b172e1264e302ac3db6c2d32303e84e753185ee182cbef9ca2f0f25R25 it wouldn't work since you use multiple sources.

Reply:
Updated.